### ISSUE TITLE
Issues with load order

### ISSUE THREAD

**Issue by SlowpokeVG** [nstlaurent/DoomLauncher#142]
Hi, after using Doom Launcher for a while I've ran into a few issues:

1. Launcher sorts files by extension before adding them into launch parameters line, this messes up load order. When I've [removed call to sort function completely](https://github.com/SlowpokeVG/DoomLauncher/commit/01595b9729beb76f148e90e2eabdabda810e7879) load order started to behave as I expected it to be. Not sure if sorting by file type is required for some ports, but I haven't encountered problems yet.

2. By default Source port's Additional files load after the file itself. I believe more logical approach is to [load them first](https://github.com/SlowpokeVG/DoomLauncher/commit/670d21e68740d11467014c9c8c65990a6d40ad78), so they wouldn't override mod's changes if there are any.

3. After adding a new mod into load order, flags from all individual files except the new one disappear, even the main file, whose load order I edit.

4. New files added into Additional Source port's files get added into many load orders unmarked in the "individual file" list. This might've been done to prevent new added files from messing up already customized load orders, but when you have more than 1k mods it becomes a problem. 

Love your launcher, thank you!

**Comment by nstlaurent** [comment]
Hi,

1. This can probably go away. It looks like it may have been there to deal with keeping file types together for -file and -deh, but it's handled better elsewhere.

2. The original use case was the 'noendgame' mod for ZDoom. This allowed ZDoom to write a save for the last level so Doom Launcher could read stats. It was assumed to be that any files with source ports/IWADS were important. If having this work the other way is really important for you, I will have to come up with an option for it. I think I will get complaints from other users if I just flip it.

3 and 4. Could you give me specific use cases? I'm not really seeing the problem, but I may not be fully understanding it and/or not using the right configuration. Thanks!

**Comment by SlowpokeVG** [comment]
2 would be neat to have at least as an option, to be sure that wad's custom stuff doesn't get overwritten by Sourceport's default mods.

Let me illustrate 3 and 4:
Here's a load order of one of my wads:
![image](https://user-images.githubusercontent.com/15838358/80923458-a4520d00-8d8c-11ea-9aa0-38bde2edaba9.png)

Here's what happens if I add, for example, "brutalist doom" mod into load order:
![image](https://user-images.githubusercontent.com/15838358/80923487-d1062480-8d8c-11ea-8caa-5eb23638f268.png)

If I enable all files again in "individual files" list I take it it saves it as "custom" load order or something, so if I add a mod ("Lowres menu" in this example) into default sourceport mod list this happens:
![image](https://user-images.githubusercontent.com/15838358/80923568-470a8b80-8d8d-11ea-8083-fd6dcf4d40b5.png)
I guess this is a good thing that it doesn't mess with previously edited load orders, but it does make it impossible to add new mod in load order for every wad.

**Comment by nstlaurent** [comment]
Thanks for the update! I will still picking away at it.

**Comment by nstlaurent** [comment]
I think I found the bug with adding brutalist doom, that one is pretty easy and I can have a build for you to test shortly. I swear I've fixed this bug before, so I'm not sure how it ended back in.

I see the the problem for adding a new file to the GZDoom, and it's not an easy one to fix. The problem is you specified you wanted to load specific files. Let's just say for simplicity it was 'brutalst.wad' and 'lights.pk3'. When you add lowres doom to GZDoom and come back the launcher knows you selected 'brutalst.wad' and 'lights.pk3' but it doesn't understand that lowres is a new mod that was added GZDoom after you saved.

Basically it does this:
1. Was there specific files selections?
2. If yes then only select the files the user wanted.
3. Else, check supported extensions for this source port and auto select files with those extensions.

I have an idea to fix this last one, but it's going to take some time.

**Comment by SlowpokeVG** [comment]
Neat, will be happy to test! 

Glad to know that you have an idea how to fix second issue too, I'm obviously in no hurry, just hope you'll fix this.

**Comment by nstlaurent** [comment]
This build should resolve the issue where you were adding brutalist doom:

https://ci.appveyor.com/api/buildjobs/disqqpb4dpitftry/artifacts/ReleaseBuild%2FDoomLauncher_2.7.0.0.zip

**Comment by SlowpokeVG** [comment]
Indeed it does. Used for a while and it seems to be working great, thanks
